src/Microsoft.Fx.Portability/ObjectModel/DiagnosticAnalyzerInfo.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System;
+
+namespace Microsoft.Fx.Portability.ObjectModel
+{
+#if FEATURE_SERIALIZABLE
+    [Serializable]
+#endif
+    public class DiagnosticAnalyzerInfo
+    {
+        public string AnalyzerName { get; set; }
+        public string Id { get; set; }
+        public ICollection<CompatibilityRangeAttribute> Attributes { get; set; }
+        public bool IsCompatibilityDiagnostic { get; set; }
+
+        public int GetIdNumber()
+        {
+            var matches = Regex.Match(Id, @"(\d+)");
```

Comment:
You should pass in the RegexOptions.Compiled flag - this will likely be called many times.